dtfs-central-api/api-tests/v1/utilisation-reports/post-upload-utilisation-report.api-test.ts

```diff
@@ -0,0 +1,210 @@
+import { wipe } from '../../wipeDB';
+import { aUtilisationReportRawCsvData, aPortalUser } from '../../../test-helpers/test-data';
+import { PostUploadUtilisationReportRequestBody } from '../../../src/v1/controllers/utilisation-report-service/post-upload-utilisation-report.controller';
+
+console.error = jest.fn();
+
+const getUrl = () => '/v1/utilisation-reports';
+
+describe(`POST ${getUrl()}`, () => {
+  const reportId = 12;
+
+  const portalUser = aPortalUser();
+  const portalUserId = portalUser._id.toString();
+
+  const aValidPayload = (): PostUploadUtilisationReportRequestBody => ({
+    reportId,
+    fileInfo: MOCK_AZURE_FILE_INFO,
+    reportData: [aUtilisationReportRawCsvData()],
+    user: { _id: portalUserId },
+  });
+
+  const aNotReceivedReport = () => UtilisationReportEntityMockBuilder.forStatus('REPORT_NOT_RECEIVED').withId(reportId).build();
+
+  beforeAll(async () => {
+    await wipe(['users']);
+    const usersCollection = await mongoDbClient.getCollection('users');
+    await usersCollection.insertOne(portalUser);
+  });
+
+  beforeEach(async () => {
+    await SqlDbHelper.deleteAll();
+    await SqlDbHelper.saveNewEntry('UtilisationReport', aNotReceivedReport());
+  });
+
+  afterAll(async () => {
+    await SqlDbHelper.deleteAll();
+  });
+
+  it('responds with a 404 (Not Found) when the report with the specified id does not exist', async () => {
+    // Arrange
+    const payload: PostUploadUtilisationReportRequestBody = { ...aValidPayload(), reportId: 999 };
+
+    // Act
+    const response = await testApi.post(payload).to(getUrl());
+
+    // Assert
+    expect(response.status).toBe(HttpStatusCode.NotFound);
+  });
+
+  it("responds with a 201 (Created) with a valid payload and sets the report status to 'PENDING_RECONCILIATION'", async () => {
+    // Act
+    const response = await testApi.post(aValidPayload()).to(getUrl());
+
+    // Assert
+    expect(response.status).toBe(HttpStatusCode.Created);
+
+    const report = await SqlDbHelper.manager.findOneByOrFail(UtilisationReportEntity, { id: reportId });
+    expect(report.status).toBe<UtilisationReportReconciliationStatus>('PENDING_RECONCILIATION');
+  });
+
+  it('creates as many fee records as there are rows in the reportData field', async () => {
+    // Arrange
+    const reportData: UtilisationReportRawCsvData[] = [
+      { ...aUtilisationReportRawCsvData() },
+      { ...aUtilisationReportRawCsvData() },
+      { ...aUtilisationReportRawCsvData() },
+    ];
+
+    const payload: PostUploadUtilisationReportRequestBody = { ...aValidPayload(), reportData };
+
+    // Act
+    const response = await testApi.post(payload).to(getUrl());
+
+    // Assert
+    expect(response.status).toBe(HttpStatusCode.Created);
+
+    const feeRecordCount = await SqlDbHelper.manager.count(FeeRecordEntity, {});
+    expect(feeRecordCount).toBe(3);
+  });
+
+  it('creates an entry in the FacilityUtilisationData table for each facility id in the report csv data', async () => {
```

Comment:
What is the intended difference between this test and `creates a new FacilityUtilisationData row using the report reportPeriod if the report data has a facility id which does not already exist` below?

Reply:
Have updated the logic and, as a result, this test